src/fhir_populator/populator.py

```diff
@@ -424,9 +442,79 @@ def get_latest_package_version(self, package_name: str) -> str:
         self.log.debug(f"Latest version: {last_version}")
         return last_version
 
+    def persistate(self, dependency_graph):
+        def copy_directory_structure(src, dst):
+            """
+            Recursively creates directories from src to dst
+            """
+            for root, dirs, files in os.walk(src):
+                for directory in dirs:
+                    src_dir = os.path.join(root, directory)
+                    dst_dir = os.path.join(dst, os.path.relpath(src_dir, src))
+                    if not os.path.exists(dst_dir):
+                        os.makedirs(dst_dir)
+
+        def copy_files_with_structure(src, dst):
+            """
+            Copies files along with the directory structure from src to dst
+            """
+            copy_directory_structure(src, dst)
+            for root, dirs, files in os.walk(src):
+                for file in files:
+                    src_file = os.path.join(root, file)
+                    dst_file = os.path.join(dst, os.path.relpath(src_file, src))
+                    shutil.copy2(src_file, dst_file)
```

Comment:
[`shutil.copytree`](https://docs.python.org/3/library/shutil.html#shutil.copytree) can copy an entire tree (and create intermediate dirs as needed, leading a call to `os.makedirs` redundant as well.